File: ReactApp/address-app-react/src/app/services/addressService.ts
import { AddressModel } from "../models/addresses/AddressModel";
import { AddressSearchModel } from "../models/addresses/AddressSearchModel";
import Service from "./service";

export default class AddressService {
    searchAddresses = async (AddressSearchFields: AddressSearchModel) => {
        try {
            return await Service.Addresses.search(AddressSearchFields);
        } 
        catch (error) {
            console.log(error);
        }

    }
    loadAddresses = async () => {
        try {
            return await Service.Addresses.list();
        } 
        catch (error) {
            console.log(error);
        }
    }

    loadAddress = async (id: string) => {
        try {
            return await Service.Addresses.details(id);
        } 
        catch (error) {
            console.log(error);
        }
    }

    saveAddress = async (Address: AddressModel) => {
        try {
            return await Service.Addresses.save(Address);
        } 
        catch (error) {
            console.log(error);
        }
    }
}

File: BlazorApp/BlazorAddressApp/BlazorAddressApp/AddressApp/Services/AddressService.cs
using AddressApp.Models.Locations;

namespace AddressApp.Services
{
    public class AddressService : BaseService
    {
        #region Base URL
        
        private string _baseURL = "https://localhost:5010/";
        
        #endregion

        #region Address

        public async Task<List<AddressSearchResultsModel>> Search(AddressSearchModel search)
        {
            return await this.PostAPIResult<List<AddressSearchResultsModel>, AddressSearchModel>(_baseURL, "addresses/", search);
        }

        public async Task<AddressModel> Get(string addressId)
        {
            return await this.GetAPIResult<AddressModel>(_baseURL, string.Format(String.Format("addresses/{0}", addressId.ToString())));
        }


        public async Task<List<string>> Save(AddressModel address)
        {
            return await this.PostAPIResult<List<string>, AddressModel>(_baseURL, "addresses/", address);
        }

        public async Task<bool> Delete(string addressId)
        {
            return await this.DeleteAPIResult(_baseURL, string.Format(String.Format("addresses/{0}", addressId.ToString())));
        }
        #endregion

    }
}